src/anyio/_backends/_asyncio.py

```diff
@@ -1922,6 +1920,8 @@ async def _call_in_runner_task(
                 self._run_tests_and_fixtures(receive_stream)
             )
 
+            self._runner_task.add_done_callback(lambda _: self._send_stream.close())
```

Comment:
it might be nicer to do:

```
async def coro():
    try:
        self._run_tests_and_fixtures(receive_stream)
     finally:
         self._send_stream.close()

self._runner_task = self.get_loop().create_task(coro())
```
?

Reply:
How about now?